query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/converters/PinotFunctionConverter.java

```diff
@@ -87,7 +87,8 @@ private String functionToStringForAvgRate(
     long aggregateIntervalInSeconds =
         (executionContext
                 .getTimeSeriesPeriod()
-                .orElseGet(() -> executionContext.getTimeRangeDuration().get()))
+                .or(executionContext::getTimeRangeDuration)
+                .orElseThrow(RuntimeException::new))
```

Comment:
`.orElseThrow()` is sufficient - we're making a specific error more generic here.